altitude_parsing.c

```diff
@@ -75,11 +90,17 @@ void parse_altitude(void){
             strncat(string, &element, 1);
         }
         else if(strlen(string) > 0){ // if we hit a line ending, and our string has a number to read:
-            altitude = strtol(string, NULL, 10);    // read the altitude from the received string
-            memset(string, 0, strlen(string));  // and clear the string so we can start again
+            altitude = strtol(string, NULL, 10);    // read the new altitude from the received string
+            memset(string, 0, strlen(string));  //clear the string so we can start parsing again
             new_altitude = true;
+            if(millis() - prev_time >= VELOCITY_ESTIMATION_INTERVAL){
```

Comment:
If `millis()` change during these code would the result be affected by much?

Reply:
a difference between line 97 and 99 might make a bit of a difference, although not much, and given the math should be quite fast, it will be rare that the value returned by millis() changes. There is far more error from the fact that we are receiving data at a pretty abysmal 9600 baud.